backend/codesirius/problems/views/problem.py

```
@@ -0,0 +1,132 @@
+            logger.info(f"Problem with ID: {pk} fetched successfully")
+        except Problem.DoesNotExist:
+            logger.warning(f"Problem with ID: {pk} not found")
+            raise NotFound("Problem not found")
+        serializer = ProblemSerializer(instance=problem)
+        return CodesiriusAPIResponse(data=serializer.data)
+
+    def put(self, request: Request, pk: int) -> CodesiriusAPIResponse:
+        """
+        Update a problem by its primary key
+        """
+        logger.info(f"Updating problem with ID: {pk}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)